### ISSUE TITLE
Fatal error, error running R command

### ISSUE THREAD

**Issue by fisherluke** [barricklab/breseq#240]
I keep getting a fatal error message which seems to be linked to R not running properly. I am very new to bioinformatics and am having a hard time trouble shooting the issue. I have updated R, bowtie2, breseq, and made sure I had all the necessary dependencies. Here is a screenshot of the error I am getting. Also, I am still getting output but as a summary.json file, not a summary.html file. What is the .json file? I would really appreciate any help!
<img width="1295" alt="Screen Shot 2020-05-27 at 1 27 07 PM" src="https://user-images.githubusercontent.com/66036977/83070804-dddf0680-a020-11ea-8dec-0cbec11c90f5.png">

**Comment by jeffreybarrick** [comment]
It looks like _breseq_ is dying and aborting before it completes making the output files.

Can you post these?
(1) The full output to the terminal of the whole breseq run.
(2) The result if you copy and paste everything after `[system]` (starting with `R`) into the terminal and run just that command.

This will hopefully help us diagnose what is happening...

**Comment by fisherluke** [comment]
Thanks for the response! Here is the full output (I skipped the POSITION: lines if that is okay)

---> bowtie2  :: version 2.3.5.1 [/home/ec2-user/miniconda3/bin/bowtie2]
---> R        :: version 4.0.0 [/home/ec2-user/miniconda3/bin/R]
+++   NOW PROCESSING Read and reference sequence file input
  READ FILE::SRR10863232_1
    Converting/filtering FASTQ file...
    Original base quality format: SANGER New format: SANGER
    Original reads: 1099694 bases: 260527131
    Filtered reads:     353 bases:     12355 (≥50.0% bases N)
    Analyzed reads: 1099341 bases: 260514776
  READ FILE::SRR10863232_2
    Converting/filtering FASTQ file...
    Original base quality format: SANGER New format: SANGER
    Original reads: 1099694 bases: 260979915
    Filtered reads:     347 bases:     12145 (≥50.0% bases N)
    Filtered reads:      18 bases:      5400 (≥90.0% same base)
    Analyzed reads: 1099329 bases: 260962370
  ::TOTAL::
    Original reads: 2199388 bases: 521507046
    Analyzed reads: 2198670 bases: 521477146
[samtools] faidx Parental_3/data/reference.fasta
  REFERENCE: NC_000913
  LENGTH: 4641652
+++   NOW PROCESSING Read alignment to reference genome
[system] bowtie2-build -q "Parental_3/data/reference.fasta" "Parental_3/02_reference_alignment/reference"
[system] bowtie2 -t --no-unal -p 4 -L 31 --ma 1 --mp 3 --np 0 --rdg 2,3 --rfg 2,3 --ignore-quals --local -i S,1,0.25 --score-min L,1,0.9 -k 2000 --reorder -x "Parental_3/02_reference_alignment/reference" -U "Parental_3/01_sequence_conversion/SRR10863232_1.converted.fastq" -S "Parental_3/02_reference_alignment/SRR10863232_1.stage1.sam" --un "Parental_3/02_reference_alignment/SRR10863232_1.stage1.unmatched.fastq"
Time loading reference: 00:00:00
Time loading forward index: 00:00:00
Time loading mirror index: 00:00:00
Multiseed full-index search: 00:01:48
1099341 reads; of these:
  1099341 (100.00%) were unpaired; of these:
    8791 (0.80%) aligned 0 times
    1066511 (97.01%) aligned exactly 1 time
    24039 (2.19%) aligned >1 times
99.20% overall alignment rate
Time searching: 00:01:48
Overall time: 00:01:48
[system] bowtie2 -t --no-unal -p 4 -L 28 --ma 1 --mp 3 --np 0 --rdg 2,3 --rfg 2,3 --ignore-quals --local -i S,1,0.25 --score-min L,6,0.2 -k 2000 --reorder -x "Parental_3/02_reference_alignment/reference" -U "Parental_3/02_reference_alignment/SRR10863232_1.stage1.unmatched.fastq" -S "Parental_3/02_reference_alignment/SRR10863232_1.stage2.matched.sam"
Time loading reference: 00:00:00
Time loading forward index: 00:00:00
Time loading mirror index: 00:00:00
Multiseed full-index search: 00:00:02
8791 reads; of these:
  8791 (100.00%) were unpaired; of these:
    512 (5.82%) aligned 0 times
    7173 (81.59%) aligned exactly 1 time
    1106 (12.58%) aligned >1 times
94.18% overall alignment rate
Time searching: 00:00:02
Overall time: 00:00:02
[system] bowtie2 -t --no-unal -p 4 -L 31 --ma 1 --mp 3 --np 0 --rdg 2,3 --rfg 2,3 --ignore-quals --local -i S,1,0.25 --score-min L,1,0.9 -k 2000 --reorder -x "Parental_3/02_reference_alignment/reference" -U "Parental_3/01_sequence_conversion/SRR10863232_2.converted.fastq" -S "Parental_3/02_reference_alignment/SRR10863232_2.stage1.sam" --un "Parental_3/02_reference_alignment/SRR10863232_2.stage1.unmatched.fastq"
Time loading reference: 00:00:00
Time loading forward index: 00:00:00
Time loading mirror index: 00:00:00
Multiseed full-index search: 00:01:55
1099329 reads; of these:
  1099329 (100.00%) were unpaired; of these:
    58781 (5.35%) aligned 0 times
    1017997 (92.60%) aligned exactly 1 time
    22551 (2.05%) aligned >1 times
94.65% overall alignment rate
Time searching: 00:01:55
Overall time: 00:01:55
[system] bowtie2 -t --no-unal -p 4 -L 28 --ma 1 --mp 3 --np 0 --rdg 2,3 --rfg 2,3 --ignore-quals --local -i S,1,0.25 --score-min L,6,0.2 -k 2000 --reorder -x "Parental_3/02_reference_alignment/reference" -U "Parental_3/02_reference_alignment/SRR10863232_2.stage1.unmatched.fastq" -S "Parental_3/02_reference_alignment/SRR10863232_2.stage2.matched.sam"
Time loading reference: 00:00:00
Time loading forward index: 00:00:00
Time loading mirror index: 00:00:00
Multiseed full-index search: 00:00:17
58781 reads; of these:
  58781 (100.00%) were unpaired; of these:
    4622 (7.86%) aligned 0 times
    50968 (86.71%) aligned exactly 1 time
    3191 (5.43%) aligned >1 times
92.14% overall alignment rate
Time searching: 00:00:17
Overall time: 00:00:17
[system] rm -f "Parental_3/02_reference_alignment/reference*"
[system] rm -f "Parental_3/02_reference_alignment/SRR10863232_1.stage1.unmatched.fastq"
[system] rm -f "Parental_3/02_reference_alignment/SRR10863232_1.stage1.sam"
[system] rm -f "Parental_3/02_reference_alignment/SRR10863232_1.stage2.matched.sam"
[system] rm -f "Parental_3/02_reference_alignment/SRR10863232_2.stage1.unmatched.fastq"
[system] rm -f "Parental_3/02_reference_alignment/SRR10863232_2.stage1.sam"
[system] rm -f "Parental_3/02_reference_alignment/SRR10863232_2.stage2.matched.sam"
+++   NOW PROCESSING Preprocessing alignments for candidate junction identification
Preprocessing alignments.
  READ FILE::SRR10863232_1
    ALIGNED READ:100000
    ALIGNED READ:200000
    ALIGNED READ:300000
    ALIGNED READ:400000
    ALIGNED READ:500000
    ALIGNED READ:600000
    ALIGNED READ:700000
    ALIGNED READ:800000
    ALIGNED READ:900000
    ALIGNED READ:1000000
  READ FILE::SRR10863232_2
    ALIGNED READ:1100000
    ALIGNED READ:1200000
    ALIGNED READ:1300000
    ALIGNED READ:1400000
    ALIGNED READ:1500000
    ALIGNED READ:1600000
    ALIGNED READ:1700000
    ALIGNED READ:1800000
    ALIGNED READ:1900000
    ALIGNED READ:2000000
    ALIGNED READ:2100000
  Summary... 
  Aligned reads:                              2193536
  Read alignments:                            2419205
  Alignments split on indels:                     527
  Reads with alignments split on indels:          325
  Split alignments:                             22742
  Reads with split alignments:                   5066
+++   NOW PROCESSING Preliminary analysis of coverage distribution
[samtools] import Parental_3/data/reference.fasta.fai Parental_3/03_candidate_junctions/best.sam Parental_3/03_candidate_junctions/best.unsorted.bam
[samtools] sort --threads 4 -o Parental_3/03_candidate_junctions/best.bam -T Parental_3/03_candidate_junctions/best.bam Parental_3/03_candidate_junctions/best.unsorted.bam
[samtools] index Parental_3/03_candidate_junctions/best.bam Parental_3/03_candidate_junctions/best.bam.bai
  REFERENCE: NC_000913
[system] R --vanilla < "/home/ec2-user/miniconda3/bin/../share/breseq/coverage_distribution.r" > "Parental_3/03_candidate_junctions/0.unique_only_coverage_distribution.tab.r.log" --args distribution_file="Parental_3/03_candidate_junctions/0.unique_only_coverage_distribution.tab" plot_file="Parental_3/03_candidate_junctions/0.coverage.pdf" deletion_propagation_pr_cutoff=2.32078e-05
[system] rm -f "Parental_3/03_candidate_junctions/best.unsorted.bam"
[system] rm -f "Parental_3/03_candidate_junctions/best.sam"
[system] rm -f "Parental_3/03_candidate_junctions/best.bam"
[system] rm -f "Parental_3/03_candidate_junctions/best.bam.bai"
[system] rm -f "Parental_3/03_candidate_junctions/0.coverage.pdf"
[system] rm -f "Parental_3/03_candidate_junctions/0.unique_only_coverage_distribution.tab"
[system] rm -f "Parental_3/03_candidate_junctions/0.unique_only_coverage_distribution.tab.r.log"
+++   NOW PROCESSING Identifying junction candidates
  READ FILE::SRR10863232_1
  READ FILE::SRR10863232_2
  Passed alignment pairs examined: 2564
  Taking top candidate junctions...
  Minimum number to keep:     100 
  Maximum number to keep:    5000 
  Maximum length to keep:  464165 bases
    Initial: Number = 292, Cumulative Length = 177997 bases
      Testing Pos Hash Score =   40, Num =      1, Length =    610
      Testing Pos Hash Score =   39, Num =      1, Length =    618
      Testing Pos Hash Score =   36, Num =      1, Length =    618
      Testing Pos Hash Score =   33, Num =      1, Length =    602
      Testing Pos Hash Score =   31, Num =      1, Length =    617
      Testing Pos Hash Score =   23, Num =      1, Length =    606
      Testing Pos Hash Score =   19, Num =      1, Length =    610
      Testing Pos Hash Score =    7, Num =      1, Length =    604
      Testing Pos Hash Score =    6, Num =      1, Length =    612
      Testing Pos Hash Score =    4, Num =      1, Length =    611
      Testing Pos Hash Score =    3, Num =      9, Length =   5521
      Testing Pos Hash Score =    2, Num =    273, Length = 166368
    Accepted: Number = 292, Pos Hash Score >= 2, Cumulative Length = 177997 bases
[samtools] faidx Parental_3/03_candidate_junctions/candidate_junction.fasta
[system] rm -f "Parental_3/03_candidate_junctions/SRR10863232_1.split.sam"
[system] rm -f "Parental_3/03_candidate_junctions/SRR10863232_2.split.sam"
+++   NOW PROCESSING Re-alignment to junction candidates
[system] bowtie2-build -q "Parental_3/03_candidate_junctions/candidate_junction.fasta" "Parental_3/04_candidate_junction_alignment/candidate_junction"
[system] bowtie2 -t --no-unal -p 4 --local  -L 31 --ma 1 --mp 3 --np 0 --rdg 2,3 --rfg 2,3 --ignore-quals --local -i S,1,0.25 --score-min L,1,0.70 -k 2000 --reorder -x "Parental_3/04_candidate_junction_alignment/candidate_junction" -U "Parental_3/01_sequence_conversion/SRR10863232_1.converted.fastq" -S "Parental_3/04_candidate_junction_alignment/SRR10863232_1.candidate_junction.sam"
Time loading reference: 00:00:00
Time loading forward index: 00:00:00
Time loading mirror index: 00:00:00
Multiseed full-index search: 00:00:33
1099341 reads; of these:
  1099341 (100.00%) were unpaired; of these:
    1075994 (97.88%) aligned 0 times
    12366 (1.12%) aligned exactly 1 time
    10981 (1.00%) aligned >1 times
2.12% overall alignment rate
Time searching: 00:00:33
Overall time: 00:00:33
[system] bowtie2 -t --no-unal -p 4 --local  -L 31 --ma 1 --mp 3 --np 0 --rdg 2,3 --rfg 2,3 --ignore-quals --local -i S,1,0.25 --score-min L,1,0.70 -k 2000 --reorder -x "Parental_3/04_candidate_junction_alignment/candidate_junction" -U "Parental_3/01_sequence_conversion/SRR10863232_2.converted.fastq" -S "Parental_3/04_candidate_junction_alignment/SRR10863232_2.candidate_junction.sam"
Time loading reference: 00:00:00
Time loading forward index: 00:00:00
Time loading mirror index: 00:00:00
Multiseed full-index search: 00:00:32
1099329 reads; of these:
  1099329 (100.00%) were unpaired; of these:
    1076401 (97.91%) aligned 0 times
    12216 (1.11%) aligned exactly 1 time
    10712 (0.97%) aligned >1 times
2.09% overall alignment rate
Time searching: 00:00:32
Overall time: 00:00:32
[system] rm -f "Parental_3/04_candidate_junction_alignment/candidate_junction*"
+++   NOW PROCESSING Resolving best read alignments
[system] rm -f "Parental_3/01_sequence_conversion/SRR10863232_1.converted.fastq"
[system] rm -f "Parental_3/01_sequence_conversion/SRR10863232_2.converted.fastq"
[system] rm -f "Parental_3/02_reference_alignment/SRR10863232_1.reference.sam"
[system] rm -f "Parental_3/02_reference_alignment/SRR10863232_2.reference.sam"
[system] rm -f "Parental_3/04_candidate_junction_alignment/SRR10863232_1.candidate_junction.sam*"
[system] rm -f "Parental_3/04_candidate_junction_alignment/SRR10863232_2.candidate_junction.sam*"
+++   NOW PROCESSING Creating BAM files
[samtools] import Parental_3/03_candidate_junctions/candidate_junction.fasta.fai Parental_3/05_alignment_correction/junction.sam Parental_3/06_bam/junction.unsorted.bam
[samtools] sort --threads 4 -o Parental_3/06_bam/junction.bam -T Parental_3/06_bam/junction.bam Parental_3/06_bam/junction.unsorted.bam
[system] rm -f "Parental_3/06_bam/junction.unsorted.bam"
[samtools] index Parental_3/06_bam/junction.bam Parental_3/06_bam/junction.bam.bai
[samtools] import Parental_3/data/reference.fasta.fai Parental_3/05_alignment_correction/reference.sam Parental_3/06_bam/reference.unsorted.bam
[samtools] sort --threads 4 -o Parental_3/data/reference.bam -T Parental_3/data/reference.bam Parental_3/06_bam/reference.unsorted.bam
[system] rm -f "Parental_3/06_bam/reference.unsorted.bam"
[samtools] index Parental_3/data/reference.bam Parental_3/data/reference.bam.bai
[system] rm -f "Parental_3/05_alignment_correction/reference.sam"
[system] rm -f "Parental_3/05_alignment_correction/junction.sam"
+++   NOW PROCESSING Tabulating error counts
  REFERENCE: NC_000913
+++   NOW PROCESSING Re-calibrating base error rates
[system] R --vanilla < "/home/ec2-user/miniconda3/bin/../share/breseq/coverage_distribution.r" > "Parental_3/07_error_calibration/0.unique_only_coverage_distribution.tab.r.log" --args distribution_file="Parental_3/07_error_calibration/0.unique_only_coverage_distribution.tab" plot_file="Parental_3/output/calibration/0.unique_coverage.pdf" deletion_propagation_pr_cutoff=2.32078e-05
[system] R --vanilla < "/home/ec2-user/miniconda3/bin/../share/breseq/plot_error_rate.r" > "Parental_3/07_error_calibration/SRR10863232_1.plot_error_rate.log" --args in_file="Parental_3/07_error_calibration/base_qual_error_prob.SRR10863232_1.tab" out_file="Parental_3/output/calibration/SRR10863232_1.error_rates.pdf" < "/home/ec2-user/miniconda3/bin/../share/breseq/plot_error_rate.r" > "Parental_3/07_error_calibration/SRR10863232_1.plot_error_rate.log"
[system] R --vanilla < "/home/ec2-user/miniconda3/bin/../share/breseq/plot_error_rate.r" > "Parental_3/07_error_calibration/SRR10863232_2.plot_error_rate.log" --args in_file="Parental_3/07_error_calibration/base_qual_error_prob.SRR10863232_2.tab" out_file="Parental_3/output/calibration/SRR10863232_2.error_rates.pdf" < "/home/ec2-user/miniconda3/bin/../share/breseq/plot_error_rate.r" > "Parental_3/07_error_calibration/SRR10863232_2.plot_error_rate.log"
+++   NOW PROCESSING Examining read alignment evidence
  REFERENCE: NC_000913
<img width="1439" alt="Screen Shot 2020-05-27 at 2 00 08 PM" src="https://user-images.githubusercontent.com/66036977/83071714-69a56280-a022-11ea-836e-60d40266e4a0.png">

Let me know if this is what you were looking for. 

Luke

**Comment by fisherluke** [comment]
Sorry, looks like this screenshot didn't make it on that post. 
<img width="1439" alt="Screen Shot 2020-05-27 at 1 56 30 PM" src="https://user-images.githubusercontent.com/66036977/83071874-a7a28680-a022-11ea-9526-233c1735b49d.png">

**Comment by jeffreybarrick** [comment]
It looks like your R is not set up to output images-which can happen sometimes on server systems. I'd recommend that you try installing R in a different what than through conda/bioconda. It looks like you are on EC2, so there should be a package manager like `yum` that you can use. You may need to uninstall or deactivate the conda/bioconda version to change the default version that gets run to the other one.

**Comment by fisherluke** [comment]
Thank you Professor Barrick, I'll give that a shot!

Luke

**Comment by fisherluke** [comment]
Hi Professor Barrick,

Thought I would post the fix here if you/anyone else is interested or has a similar error. I had to uninstall R (per your recommendation) which uninstalled breseq and bowtie2. I guess this occurred because I installed breseq via bioconda which downloaded R and bowtie concomitantly (although I am not positive if this is the reason). I reinstalled R and bowtie2 using [sudo yum install]. Then I used wget to download the unix version you posted (for aws), added it to my path, and done! Finally got a successful summary. Thanks again for your help!
